kaggle_environments/envs/lux_ai_2022/luxai2022/env.py

```diff
@@ -0,0 +1,618 @@
+                factory_id = f"factory_{self.state.board.factory_occupancy_map[transfer_pos.y, transfer_pos.x]}"
+                if factory_id in self.state.factories[unit.team.agent]:
+                    factory = self.state.factories[unit.team.agent][factory_id]
+                    # add resources to target. This will waste (transfer_amount - actually_transferred) resources
+                    actually_transferred = factory.add_resource(transfer_action.resource, transfer_action.transfer_amount)
+                elif units_there is not None:
+                    assert len(units_there) == 1
+                    target_unit = units_there[0]
+                    # add resources to target. This will waste (transfer_amount - actually_transferred) resources
+                    actually_transferred = target_unit.add_resource(transfer_action.resource, transfer_amount)
+
+            for unit, pickup_action in actions_by_type["pickup"]:
+                pickup_action: PickupAction
+                factory = self.state.board.get_factory_at(unit.pos)
+                pickup_amount = factory.sub_resource(pickup_action.resource, pickup_action.pickup_amount)
+                # may waste resources if tried to pickup more than one can hold.
+                actually_pickedup = unit.add_resource(pickup_action.resource, pickup_amount)
+
+            for unit, dig_action in actions_by_type["dig"]:
+                dig_action: DigAction
+                if self.state.board.rubble[unit.pos.y, unit.pos.x] > 0:
+                    self.state.board.rubble[unit.pos.y, unit.pos.x] = max(self.state.board.rubble[unit.pos.y, unit.pos.x] - unit.unit_cfg.DIG_RUBBLE_REMOVED, 0)
+                elif self.state.board.lichen[unit.pos.y, unit.pos.x] > 0:
+                    self.state.board.lichen[unit.pos.y, unit.pos.x] = max(self.state.board.lichen[unit.pos.y, unit.pos.x] - unit.unit_cfg.DIG_LICHEN_REMOVED, 0)
+                elif self.state.board.ice[unit.pos.y, unit.pos.x] > 0:
+                    unit.add_resource(0, unit.unit_cfg.DIG_RESOURCE_GAIN)
+                elif self.state.board.ore[unit.pos.y, unit.pos.x] > 0:
+                    unit.add_resource(1, unit.unit_cfg.DIG_RESOURCE_GAIN)
+                unit.power -= math.ceil(self.state.env_cfg.ROBOTS[unit.unit_type.name].DIG_COST * weather_cfg["power_loss_factor"])
+
+            for unit, self_destruct_action in actions_by_type["self_destruct"]:
+                unit: Unit
+                self_destruct_action: SelfDestructAction
+                pos_hash = self.state.board.pos_hash(unit.pos)
+                del self.state.board.units_map[pos_hash]
+                self.destroy_unit(unit)
+
+            for factory, factory_build_action in actions_by_type["factory_build"]:
+                factory: Factory
+                factory_build_action: FactoryBuildAction
+                team = self.state.teams[factory.team.agent]
+                self.add_unit(
```

Comment:
do you need a check to see if a unit is in the spawn point? or does it get taken care of later (if it doesn't move) in collisions?

Reply:
Collisions should take care of it

Reply:
Would be good to mention in docs